### PR TITLE
:memo: replace all instances of Web3 with w3

### PR DESCRIPTION
**Pull request by Harmouch101**
Signed-off-by: Harmouch101 <eng.mahmoudharmouch@gmail.com>

### What was wrong?

Related to Issue #1183

### How was it fixed?
Using the power of `vs code` IntelliSense along with the search and replace(case sensitive) feature to replace each `web3` instance variable name with `w3` to avoid confusion with the name of the `web3 module`.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

:guide_dog:

### CODE REVIEW

**Comment by Harmouch101** [comment]
Good lord, it passes. I feel pain. Awaiting reviews from @pipermerriam, @kclowes, @fselmo...

File: tests/core/contracts/test_extracting_event_data.py [hunk 1]
```diff
@@ -750,7 +750,7 @@ def test_receipt_processing_with_no_flag(
 
 
 def test_single_log_processing_with_errors(
-        web3,
+        w3,
```
Comment (fselmo) [inline tests/core/contracts/test_extracting_event_data.py:753]: We can probably remove these arguments altogether. Doesn't seem like we use the web3 instance in these tests.

**Review by fselmo — COMMENTED** [review]
I left some nits here that I can take care of too. I'm going to pull in the branch in a bit and do a quick search to see if we missed anything but this PR is looking good. Thanks for taking the time! Really appreciate it. 😅

File: tests/core/contracts/test_extracting_event_data.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harmouch101) [reply tests/core/contracts/test_extracting_event_data.py]: Yeah, I was wondering why this fixture was passed and not used.

**Review by kclowes — APPROVED** [review]
@Harmouch101 thank you for doing all this, and thanks @fselmo for the review. I looked through too, and it looks good to me!

**Pull request merged**